ledger-core/virtual/handlers/vdelegatedcallrequest.go

```diff
@@ -192,26 +197,35 @@ func (s *SMVDelegatedCallRequest) stepProcessRequest(ctx smachine.ExecutionConte
 }
 
 func (s *SMVDelegatedCallRequest) stepBuildResponse(ctx smachine.ExecutionContext) smachine.StateUpdate {
-	target := s.Meta.Sender
+	var (
+		delegateTo   = s.Meta.Sender
+		callee       = s.Payload.Callee.GetValue()
+		pn           = s.pulseSlot.PulseNumber()
+		callOutgoing = s.Payload.CallOutgoing.GetValue()
+	)
 
-	token := s.authenticationService.GetCallDelegationToken(s.Payload.CallOutgoing, s.Meta.Sender, s.pulseSlot.PulseData().PulseNumber, s.Payload.Callee)
+	token := s.authenticationService.GetCallDelegationToken(callOutgoing, delegateTo, pn, callee)
 
-	response := payload.VDelegatedCallResponse{
+	response := rms.VDelegatedCallResponse{
 		Callee:                 s.Payload.Callee,
 		CallIncoming:           s.Payload.CallIncoming,
 		ResponseDelegationSpec: token,
 	}
 
 	s.messageSender.PrepareAsync(ctx, func(goCtx context.Context, svc messagesender.Service) smachine.AsyncResultFunc {
-		err := svc.SendTarget(goCtx, &response, target)
+		err := svc.SendTarget(goCtx, &response, delegateTo)
 		return func(ctx smachine.AsyncResultContext) {
 			if err != nil {
 				ctx.Log().Error(struct {
-					*log.Msg `txt:"failed to send VDelegatedCallResponse"`
-					Object   string
-					Request  string
-					target   string
-				}{Object: s.Payload.Callee.String(), Request: s.Payload.String(), target: target.String()}, err)
+					*log.Msg   `txt:"failed to send VDelegatedCallResponse"`
+					Object     reference.Global
+					DelegateTo reference.Global
+					Request    string
+				}{
+					Object:     callee,
+					DelegateTo: delegateTo,
+					Request:    s.Payload.String(),
```

Comment:
is it really necessary to stringify whole Payload?

Reply:
and do it here?

Reply:
I don't see any reason for that, I think it's some garbage that was left after debugging

Reply:
fixed in https://github.com/insolar/assured-ledger/pull/766